cmd/gctsRollback.go

```diff
@@ -0,0 +1,324 @@
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := rollback(&config, telemetryData, &c, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func rollback(config *gctsRollbackOptions, telemetryData *telemetry.CustomData, command execRunner, httpClient piperhttp.Sender) error {
+
+	cookieJar, cookieErr := cookiejar.New(nil)
+	if cookieErr != nil {
+		return cookieErr
+	}
+	clientOptions := piperhttp.ClientOptions{
+		CookieJar: cookieJar,
+		Username:  config.Username,
+		Password:  config.Password,
+	}
+	httpClient.SetOptions(clientOptions)
+
+	repoInfo, err := getRepoInfo(config, telemetryData, httpClient)
+	if err != nil {
+		return errors.Wrap(err, "could not get local repository data")
+	}
+
+	if repoInfo.Result.URL == "" {
+		return errors.Errorf("no remote repository URL configured")
+	}
+
+	parsedURL, err := url.Parse(repoInfo.Result.URL)
+	if err != nil {
+		return errors.Wrap(err, "could not parse remote repository URL as valid URL")
+	}
+
+	var deployParams []string
+
+	if config.Commit != "" {
+		log.Entry().Infof("rolling back to specified commit %v", config.Commit)
+
+		deployParams = []string{"gctsDeploy", "--username", config.Username, "--password", config.Password, "--host", config.Host, "--client", config.Client, "--repository", config.Repository, "--commit", config.Commit}
+
+	} else if parsedURL.Host == "github.com" {
+		log.Entry().Info("Remote repository domain is 'github.com'. Trying to rollback to last commit with status 'success'.")
+
+		commitList, err := getCommits(config, telemetryData, httpClient)
+		if err != nil {
+			return errors.Wrap(err, "could not get repository commits")
+		}
+
+		successCommit, err := getLastSuccessfullCommit(config, telemetryData, httpClient, parsedURL, commitList)
+		if err != nil {
+			return errors.Wrap(err, "could not determine successfull commit")
+		}
+
+		deployParams = []string{"gctsDeploy", "--username", config.Username, "--password", config.Password, "--host", config.Host, "--client", config.Client, "--repository", config.Repository, "--commit", successCommit}
+
+	} else {
+		repoHistory, err := getRepoHistory(config, telemetryData, httpClient)
+		if err != nil {
+			return errors.Wrap(err, "could not retrieve repository commit history")
+		}
+		if repoHistory.Result[0].FromCommit != "" {
+			log.Entry().WithField("repository", config.Repository).Infof("rolling back to last active commit %v", repoHistory.Result[0].FromCommit)
+			deployParams = []string{"gctsDeploy", "--username", config.Username, "--password", config.Password, "--host", config.Host, "--client", config.Client, "--repository", config.Repository, "--commit", repoHistory.Result[0].FromCommit}
+		} else {
+			return errors.Errorf("no commit to rollback to (fromCommit) could be identified from the repository commit history")
+		}
+	}
+
+	deployErr := command.RunExecutable("./piper", deployParams...)
```

Comment:
why calling `piper` here?
We should rather call the respective API ...

Reply:
@CCFenner  what do you think?

Reply:
I want to reuse the "gctsDeploy" step. Is there a better way to do this?

Reply:
Sorry, just saw that the answer frommstes side her is pending ...

Ideally, you just call the respective go coding of the other step and not call the piper binary itself.
We use similar pattern for mavenExecute calls from other steps.
Maybe you want to have a look there.

Reply:
removed calling piper binary